Parse/PFObject.m

```diff
@@ -2317,6 +2317,36 @@ - (void)removeObjectForKey:(NSString *)key {
     }
 }
 
+- (void)revert {
+    @synchronized (lock) {
+        if ([self isDirty]) {
+            PFOperationSet *changes = [self unsavedChanges];
+            [changes enumerateKeysAndObjectsUsingBlock:^(id key, id obj, BOOL *stop) {
+                [_availableKeys removeObject:key];
+            }];
+            
+            [operationSetQueue removeLastObject];
+            [self rebuildEstimatedData];
+            [self checkpointAllMutableContainers];
+        }
+    }
+}
+
+-(void)revertObjectForKey:(NSString *)key {
+    @synchronized (lock) {
+        if ([self isDirtyForKey:key]) {
+            NSUInteger index = [operationSetQueue indexOfObject:key];
+            [operationSetQueue removeObjectAtIndex:index];
```

Comment:
2338-2339 - I don't think you need it at this point, since it won't find the index, as well as `removeObjectAtIndex:NSNotFound` (the index returned when it doesn't find anything) - will simply crash the app.